src/Components/Modals/ActivationKeyWizard.js

```diff
@@ -62,20 +69,23 @@ const CreateActivationKeyWizard = ({
   const { data: activationKeys } = useActivationKeys();
   const { addSuccessNotification, addErrorNotification } = useNotifications();
   const [name, setName] = useState('');
-  const [description, setDescription] = useState();
+  const [description, setDescription] = useState(
+    activationKey?.description || ''
+  );
   const [workload, setWorkload] = useState(workloadOptions[0]);
   const [extendedReleaseProduct, setExtendedReleaseProduct] = useState('');
   const [extendedReleaseVersion, setExtendedReleaseVersion] = useState('');
   const [extendedReleaseRepositories, setExtendedReleaseRepositories] =
     useState([]);
-  const [role, setRole] = useState('');
-  const [sla, setSla] = useState('');
-  const [usage, setUsage] = useState('');
+  const [role, setRole] = useState(activationKey?.role);
+  const [sla, setSla] = useState(activationKey?.serviceLevel);
+  const [usage, setUsage] = useState(activationKey?.usage);
   const [isConfirmClose, setIsConfirmClose] = useState(false);
   const [shouldConfirmClose, setShouldConfirmClose] = useState(false);
   const [currentStep, setCurrentStep] = useState(0);
   const keyNames = activationKeys?.map((key) => key.name) || [];
   const nameIsValid = nameValidator(name, keyNames);
+  const isEditMode = activationKey !== undefined;
```

Comment:
Polish: Prefer passing this in as a prop when this component is called. Inferring based on the presence of an activation key should in theory mean the same thing, but an explicitly passed prop is clearer.